Sources/BrowserServicesKit/Suggestions/SuggestionLoading.swift

```diff
@@ -21,141 +21,92 @@ import Foundation
 public protocol SuggestionLoading: AnyObject {
 
     func getSuggestions(query: Query,
-                        maximum: Int,
-                        completion: @escaping ([Suggestion]?, Error?) -> Void)
+                        completion: @escaping (SuggestionResult?, Error?) -> Void)
```

Comment:
Maybe let's switch to Result<SuggestionResult, SuggestionLoaderError> completion argument? It seems more explicit to me

Reply:
Please look at `completion(result, apiError)` on the line 94. If the fetching of remote suggestions fails, SuggestionResult (from local data) and Error are returned. `Result` can’t be used because is either `success` or `failure`.